docs/source/agent-framework/web-framework.rst

```diff
@@ -54,19 +101,41 @@ JSON endpoints allows an agent to serve data responses to specific queries from
         self.vip.web.register_endpoint(r'/vc/jsonrpc', self.jsonrpc)
 
 
-Websocket endpoints allow bi-directional communication between the client and the server.  Client connections can be authenticated during the opening of a websocket through the response of an open callback.
+Websocket
+~~~~~~~~~
+
+Websocket endpoints allow bi-directional communication between the client and the server.
+Client connections can be authenticated during the opening of a websocket through the response of an open callback.
 
 
 .. code-block:: python
 
-    def _open_authenticate_ws_endpoint(self, fromip, endpoint):
+    def open_authenticate_ws_endpoint(self, fromip, endpoint):
         """
-        A client attempted to open an endpoint to the server.
-
-        Return True or False if the endpoint should be allowed.
-
-        :rtype: bool
+        Callback method from when websockets are opened.  The endpoine must
+        be '/' delimited with the second to last section being the session
```

Comment:
seems like there is a lot of things going on in this example that aren't clear and that really don't fit.  The authentication of vc is really different than the way agents should do it.

The holding of a session is also something that should probably be broken out into an "authentication section" if there isn't one there should be